### DISCUSSION TITLE
Initialization is done when rebooting after setting

### DISCUSSION THREAD

**Discussion by js-midi** [shanteacontrols/OpenDeck#51] [Q&A]
All values will be initialized when booting after setting in the official Open Deck Board 2.0.
Preset button is on
When rebooting, all values are initialized.

**Comment by paradajz** [comment]
Ehm... What?

**Reply by js-midi** [reply]
If the power is turned off after setting the configuration, the setting cannot be maintained.
Reinitialized

**Reply by js-midi** [reply]
How do I clear the entire flash memory of the MCU?

**Reply by js-midi** [reply]
And please tell me how to flash again

**Reply by paradajz** [reply]
I am sorry but I do not understand what are you talking about. 

_All values will be initialized when booting after setting_
After setting what?

_Preset button is on_
What does this mean?

**Reply by js-midi** [reply]
In firmware version 5.20, the preset values ​​are maintained.
However, firmware version 5.42 cannot be maintained on reboot.

**Reply by js-midi** [reply]
I'm sorry for my lack of English proficiency

**Reply by paradajz** [reply]
Send me a video of what you're doing, what you're expecting and what's happening. Several boards I have around here preserve the settings just fine. Also, my unit tests confirm this.

**Comment by js-midi** [comment]
https://youtu.be/8jlLpHNSIAw

https://youtu.be/9iSKHuyrk4s

**Reply by js-midi** [reply]
There was no problem with arduino,discovery boards

**Reply by paradajz** [reply]
I don't know what's going on. Did you try performing factory reset through the UI?

**Reply by js-midi** [reply]
No, I just turned on the power again

**Reply by paradajz** [reply]
Try it then. 
![image](https://user-images.githubusercontent.com/2544094/112681795-7cb9d680-8e6f-11eb-89dd-01c136bf961a.png)

**Reply by js-midi** [reply]
플래시 메모리를 지운후 다시 플래싱을 해야될거같은데
방법을 잘 모르겠어~
내가 가지고있는 st-sink/v2로 가능할까?

**Reply by js-midi** [reply]
st-link/v2

**Reply by paradajz** [reply]
???

**Reply by js-midi** [reply]
Even after factory reset, the same problem

**Reply by paradajz** [reply]
So you want to flash it again with ST Link, is that correct?

**Reply by js-midi** [reply]
There is the same error from version 5.31.

**Reply by js-midi** [reply]
yes

**Reply by paradajz** [reply]
CONN8 on OpenDeck board contains SWDIO and SWCLK pins (above the AH1 header)

![image](https://user-images.githubusercontent.com/2544094/112683677-3a45c900-8e72-11eb-9efe-0eb0a33bb433.png)

Left is SWDIO, right is SWDCLK.

Use ground connection from analog headers. You can use STM32CubeProgrammer to wipe out entire flash and then you can use  this hex file to reflash it:

https://github.com/paradajz/OpenDeck/releases/download/v5.4.2/opendeck2.hex

**Reply by js-midi** [reply]
vcc?

**Reply by paradajz** [reply]
not needed

**Reply by paradajz** [reply]
just keep the board connected to USB

**Reply by js-midi** [reply]
ok

**Reply by js-midi** [reply]
error :NO STM32 target found!

**Reply by paradajz** [reply]
send me a picture of your connections

**Reply by paradajz** [reply]
![image](https://user-images.githubusercontent.com/2544094/112685998-a0801b00-8e75-11eb-9bbc-1d79581b0446.png)

![20210326_205613](https://user-images.githubusercontent.com/2544094/112686197-ef2db500-8e75-11eb-9e86-9a4b19fc7e7e.jpg)

**Reply by paradajz** [reply]
Maybe your ST-Link requires VCC. You can connect that to +3v3, also on analog headers.

**Reply by js-midi** [reply]
I tried programming but it failed.
The board appears to be damaged.

**Reply by paradajz** [reply]
Which error are you getting?

**Reply by js-midi** [reply]
Firmware cannot be flashed.
There seems to be some damage to the board.
I think I need to replace the mcu.

**Reply by js-midi** [reply]
After purchasing the MCU, I plan to replace it and try again.
Thank you for your help.
I'll ask for help again later.

**Reply by js-midi** [reply]
error:failed to erase memory
